### PR TITLE
add api timeout attr

### PR DESCRIPTION
**Pull request by shortdudey123**
Sometimes the Sumologic API is slow to respond.  This allows optional timeouts with back off retries on the API calls.

### CODE REVIEW

**Comment by shortdudey123** [comment]
Rebased (to pick up updated cops and gems) and extracted the http call and the timeout logic to 2 different methods

**Comment by shortdudey123** [comment]
Rebased to fix merge conflict

**Comment by shortdudey123** [comment]
squashed commits

File: libraries/helper.rb [hunk 1]
```diff
@@ -4,22 +4,24 @@
...
 
   class Collector
-    attr_reader :name, :api_username, :api_password
+    attr_reader :name, :api_username, :api_password, :api_timeout
 
     def initialize(opts = {})
       @name = opts[:name]
       @api_username = opts[:api_username]
       @api_password = opts[:api_password]
+      @api_timeout = opts.fetch(:api_timeout, nil)
```
Comment (ranjib) [inline libraries/helper.rb]:
```ruby
opts[:api_password] || 60
```
or some other default timeout?

Comment (shortdudey123) [inline libraries/helper.rb]: Using `nil` disables the timeout by default for backwards compatibility.  Should we enforce an actual seconds default?

Comment (ranjib) [inline libraries/helper.rb]: we should i think it helps predicting the behavior.

File: spec/helper_spec.rb [hunk 2]
```diff
@@ -60,7 +68,7 @@
       .with(body: "{\"source\":{\"foo\":\"bar\"}}")
       .to_return(status: 200, body: '{}', headers: {})
 
-    collector.add_source!(foo: :bar)
+    collector.add_source!({ foo: :bar })
```
Comment (ranjib) [inline spec/helper_spec.rb]: why curly braces?

File: libraries/helper.rb (hunk 1 above)
Comment (shortdudey123) [inline libraries/helper.rb]: Fair point
Looks like the [default timeout](http://ruby-doc.org/stdlib-2.2.0/libdoc/net/http/rdoc/Net/HTTP.html#attribute-i-read_timeout) for Net::HTTP is 60 seconds so i will use that

File: spec/helper_spec.rb (hunk 2 above)
Comment (shortdudey123) [inline spec/helper_spec.rb]: added them for something and forgot to remove them, reverting this line back

**Pull request merged**